Source/WebCore/page/ElementTargetingController.cpp

```diff
@@ -860,6 +861,123 @@ Vector<TargetedElementInfo> ElementTargetingController::findTargets(TargetedElem
     return extractTargets(WTFMove(nodes), WTFMove(innerElement), request.canIncludeNearbyElements);
 }
 
+void ElementTargetingController::topologicallySortElementsHelper(ElementIdentifier currentElementID, Vector<ElementIdentifier>& depthSortedIDs, HashSet<ElementIdentifier>& processingIDs, HashSet<ElementIdentifier>& unprocessedIDs, const HashMap<ElementIdentifier, HashSet<ElementIdentifier>>& elementIDToOccludedElementIDs)
+{
+    if (!unprocessedIDs.contains(currentElementID) && !processingIDs.contains(currentElementID))
+        return;
+
+    if (processingIDs.contains(currentElementID)) {
+        ASSERT_NOT_REACHED();
+        return;
+    }
```

Comment:
Nit - I think you can skip 1 redundant hash lookup by rearranging the early returns:

```suggestion
    if (processingIDs.contains(currentElementID)) {
        ASSERT_NOT_REACHED();
        return;
    }

    if (!unprocessedIDs.contains(currentElementID))
        return;
```